test/Google.Api.Gax.Tests/CommonSettingsTest.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Xunit;
+
+namespace Google.Api.Gax.Tests
+{
+    public class CommonSettingsTest
+    {
+        class TestSettings : CommonSettings { }
+
+        [Fact]
+        public void DefaultToNulls()
+        {
+            var settings = new TestSettings();
+            Assert.Null(settings.Timeout);
+            Assert.Null(settings.Clock);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(1)]
+        public void Timeout_Valid(int? millis)
+        {
+            var timeout = millis == null ? default(TimeSpan?) : TimeSpan.FromMilliseconds(millis.Value);
+            var settings = new TestSettings { Timeout = timeout };
+            Assert.Equal(timeout, settings.Timeout);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public void Timeout_Invalid(int? millis)
+        {
+            var timeout = millis == null ? default(TimeSpan?) : TimeSpan.FromMilliseconds(millis.Value);
+            Assert.Throws<ArgumentOutOfRangeException>(() => new TestSettings { Timeout = timeout });
+        }
+
```

Comment:
Cloning test, possibly with TestSettings overriding it in the expected way (as much for documentation as anything)?